man/coap_pdu_setup.txt.in

```diff
@@ -0,0 +1,389 @@
+the options need to be added in the correct numerical option order as they are 
+stored in the PDU using relative numeric offsets from the previous option 
+number.
+
+There are option list functions available where options can be added to a 
+chained list of options and then the chain list is sorted and then be added to 
+the PDU.
+
+Typically for clients, when creating a request, the PDU needs to be creaed 
+before filling it with the appropriate information.
+
+Typically with a server, the response PDU, with the optional token already added
+in, will already be created before the response handler is called, and the 
+response PDU will need to be updated as appropriate starting with the optional 
+options.  Note that updating the response pdu's code variable will cause the 
+response pdu to get transmitted.  If code does not get updated, then the 
+response pdu is freed off by the underlying library.
+
+The *coap_pdu_init*() function returns a newly created _PDU_ of type 
+_coap_pdu_t_*.  _type_ is one of the following
+
+[horizontal]
+*COAP_MESSAGE_CON*::
+Set the _PDU_ to be of type confirmable.
+*COAP_MESSAGE_NON*::
+Set the _PDU_ to be of type non-confirmable.
+*COAP_MESSAGE_ACK*::
+Set the _PDU_ to be of type acknowledge (for internal use).
+*COAP_MESSAGE_RST*::
+Set the _PDU_ to be of type reset.
+
+The _code_ is one of the following
+
+[horizontal]
+*COAP_REQUEST_GET*::
+Set the _PDU_ request to be of type GET.
+*COAP_REQUEST_POST*::
+Set the _PDU_ request to be of type POST.
+*COAP_REQUEST_PUT*::
+Set the _PDU_ request to be of type PUT.
+*COAP_REQUEST_DELETE*::
+Set the _PDU_ request to be of type DELETE.
+*COAP_REQUEST_FETCH*::
+Set the _PDU_ request to be of type FETCH.
+*COAP_REQUEST_PATCH*::
+Set the _PDU_ request to be of type PATCH.
+*COAP_REQUEST_IPATCH*::
+Set the _PDU_ request to be of type IPATCH.
+
+The _message_id_ must be unique per request (which is not the same as the
+token), and must not be reused within EXCHANGE_LIFETIME (usually 247 seconds).
+To automate this, the function coap_new_message_id(session) should be called.
+
+The _max_size_ parameter defines the maximum size of a _PDU_ and is usually 
+determined by calling coap_session_max_pdu_size(session); 
+
+The *coap_add_token*() function adds in the specified token's _data_ of length 
+_length_ to the PDU _pdu_.  The maximum length of the token is 8 bytes.
+Adding the token must be done before any options or data are added. 
+This function must only be called once per _pdu_, and must not be called
+in the appropriate response handler.
+
+The *coap_new_optlist*() function returns a newly created _optlist_ entry of
+type _coap_optlist_t_*.  The _number_ specifies which CoAP option is to be 
+used, and is one of the COAP_OPTION_* definitions.  The _length_ is the length 
+of the data of the option, and _data_ points to the content of the option.
+
+The following is the current list of options with their numeric value
```

Comment:
Maybe add a reference to the [CoAP option number registry](https://www.iana.org/assignments/core-parameters/core-parameters.xhtml#option-numbers) here?
(Or, better, in the FURTHER INFORMATION section.)

Reply:
Fixed in Update